Kernel/FileSystem/DevFS.h

```diff
@@ -113,7 +113,7 @@ class DevFSDeviceInode : public DevFSInode {
     virtual ssize_t write_bytes(off_t, ssize_t, const UserOrKernelBuffer& buffer, FileDescription*) override;
     virtual KResult chown(uid_t, gid_t) override;
 
-    NonnullRefPtr<Device> m_attached_device;
+    const Device* m_attached_device;
```

Comment:
- Can `m_attached_device` now become nullptr? I don't think so, but in that case all accesses would now require checks; for example, `read_bytes` would crash otherwise.
- If it's guaranteed non-null, it should be a reference, NonnullOwnPtr, or NonnullRefPtr. Since `Device` is a `RefCounted` (through `File`), NonnullRefPtr makes the most sense.

So, what does this achieve? Note that your check `if (match->m_attached_device == &device)` can also be done for NonnullRefPtr.

Reply:
It is guaranteed to be non-null. Smart pointers aren't helpful here since the `Device` constructor and destructor ensure that a `DevFSDeviceInode` and its accompanying `Device` have the exact same lifetime. A reference would probably be better than a pointer, however.